src/mimeparser.rs

```diff
@@ -571,7 +571,7 @@ impl MimeMessage {
             decrypting_failed: mail.is_err(),
 
             // only non-empty if it was a valid autocrypt message
-            signatures,
+            signature: signatures.into_iter().next(),
```

Comment:
Probably we must take the innermost signature because in `merge_headers()` we prefer protected headers from the inner part (we drop outer protected headers at all).